src/libraries/System.Net.Security/src/System/Net/Security/SslStreamPal.Android.cs

```diff
@@ -72,7 +74,50 @@ public static void VerifyPackageInfo()
             resultSize = 0;
             Debug.Assert(input.Length > 0, $"{nameof(input.Length)} > 0 since {nameof(CanEncryptEmptyMessage)} is false");
 
-            throw new NotImplementedException(nameof(EncryptMessage));
+            try
+            {
+                SafeDeleteSslContext sslContext = (SafeDeleteSslContext)securityContext;
+                SafeSslHandle sslHandle = sslContext.SslContext;
+
+                PAL_SSLStreamStatus ret;
+                unsafe
+                {
+                    MemoryHandle memHandle = input.Pin();
+                    try
+                    {
+                        ret = Interop.AndroidCrypto.SSLStreamWrite(sslHandle, (byte*)memHandle.Pointer, input.Length);
+                    }
+                    finally
+                    {
+                        memHandle.Dispose();
+                    }
+                }
```

Comment:
Since there's only the one operation with the memory pinned can we move this unsafe block to the Interop method and call it from here with only safe code?